### PR TITLE
HDS-2325: Fix typos and clean up texts

### PR DESCRIPTION
**Pull request by NikoHelle**
## Description

Docs improvements

## Related Issue

Closes [HDS-2325](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2325

## How Has This Been Tested?

Visually

## Demos:
Links to demos are in the comments

## Add to changelog
Nothing new to users

[HDS-2325]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-2325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/components/header/index.mdx [hunk 1]
```diff
@@ -381,13 +381,13 @@ Any ActionBarItem with a functionality that opens a dropdown, must have an indic
       onMenuClick={(e) => e.stopPropagation()}
     >
       <Header.ActionBarItem label="Ilmoitukset" icon={<IconBell />} id="action-bar-notifications">
-        <Header.ActionBarSubItem label="Ei uusi ilmoituksia." />
+        <Header.ActionBarSubItem label="Ei uusi ilmoituksia" />
```
Comment (mrTuomoK) [inline site/src/docs/components/header/index.mdx:384]: Hmm... shouldn't it be "Ei uusiA ilmoituksia" ? :D :D

**Review by mrTuomoK — CHANGES_REQUESTED** [review]
All good, just the minor "uusi" -> "uusia" todo 👍

File: site/src/docs/components/header/index.mdx (hunk 1 above)
Comment (NikoHelle) [reply site/src/docs/components/header/index.mdx]: Jep! 😄

**Review by mrTuomoK — APPROVED** [review]
All good 👍

**Pull request merged**